spec.md

```diff
@@ -3907,33 +3907,262 @@ said to have "internal" convention.
+
+port = ( "input" | "output" ) , id , ":" , (type | type_property) , [ info ] ;
+type_param = int | string_dq | string_sq ;
+type_property = "Integer" ;
+
+(* Statements *)
+statement =
+    circuit_component
+  | connectlike
+  | conditional
+  | command
+  | group
+  | skip ;
+
+(* Circuit Components *)
+circuit_component =
+    circuit_component_node
+  | circuit_component_wire
+  | circuit_component_reg
+  | circuit_component_inst
+  | circuit_component_mem ;
+
+circuit_component_node = "node" , id , "=" , expr , [ info ] ;
+circuit_component_wire = "wire" , id , ":" , type , [ info ] ;
+circuit_component_inst = "inst" , id , "of" , id , [ info ] ;
+
+conditional_reg =
+    "reg" , id , ":" , type , expr , [ info ]
+  | "regreset" , id , ":" , type , "," , expr , "," , expr , "," , expr , [info] ;
+
+circuit_component_mem =
+  "mem" , id , ":" , [ info ] , newline , indent ,
+    "data-type" , "=>" , type , newline ,
+    "depth" , "=>" , int , newline ,
+    "read-latency" , "=>" , int , newline ,
+    "write-latency" , "=>" , int , newline ,
+    "read-under-write" , "=>" , read_under_write , newline ,
+    { "reader" , "=>" , id , newline } ,
+    { "writer" , "=>" , id , newline } ,
+    { "readwriter" , "=>" , id , newline } ,
+  dedent ;
+
+read_under_write =  "old" | "new" | "undefined" ;
+
+(* Connect-like Statements *)
+connectlike =
+    "connect" , reference , "," , expr , [ info ]
+  | "invalidate" , reference , [ info ]
+  | "attach" , "(" , reference , { "," ,  reference } , ")" , [ info ]
+  | "define" , reference_static , "=" , expr_probe , [ info ]
+  | "propassign" , reference_static , "," , property_expr , [ info ] ;
+
+(* Conditional Statements *)
+conditional =
+    conditional_when
+  | conditional_match ;
+
+conditional_when =
+  "when" , expr , ":" [ info ] , newline ,
+    indent , statement, { statement } , dedent ,
+  [ "else" , ":" , indent , statement, { statement } , dedent ] ;
+
+conditional_match =
+  "match" , expr , ":" , [ info ] , newline ,
+  [ indent , { conditional_match_branch } , dedent ] ;
+
+conditional_match_branch =
+  id , [ "(" , id , ")" ] , ":" , newline ,
+  [ indent , { statement } , dedent ] ;
+
+(* Command Statements *)
+command =
+    "stop" , "(" , expr , "," , expr , "," , int , ")" , [ info ]
+  | "force" , "(" , expr , "," , expr , "," , expr_probe , "," , expr , ")"
+  | "force_initial" , "(" , expr_probe , "," , expr , ")"
+  | "release" , "(" , expr , "," , expr , "," , expr_probe , ")"
+  | "release_initial" , "(" , expr_probe , ")"
+  | "printf" , "(" ,
+        expr , "," ,
+        expr , "," ,
+        string_dq ,
+        { "," , expr }
+    , ")" ,
+    [ ":" , id ] , [ info ] ;
+
+(* Group Statement *)
+group =
+  "group" , id , "of" , id , ":" , [ info ] , newline , indent ,
+    { port , newline } ,
+    { statement , newline } ,
+  dedent ;
+
+(* Skip Statement *)
+skip = "skip" , [ info ] ;
+
+(* References *)
+reference =
+    reference_static
```

Comment:
I think this may be pre-existing, but I think this makes clear a bug?

Specifically, this doesn't appear to allow a subaccess in the middle of a chain of static referencing, specifically with trailing field (subfield) ?

Reply:
Good catch.

This looks like it could be a bit tricky to resolve. I'll put this in my todos to tackle, and leave it for now.